auth/auth.go

```diff
@@ -160,6 +161,53 @@ func (auth *Authenticator) GetRoleIncDeleted(name string) (Role, error) {
 	return role, err
 }
 
+func (auth *Authenticator) InheritedChannels(user User, princ Principal) error {
+	cumulativeChannels := ch.TimedSet{}
+	roles := make([]Role, 0, len(user.RoleNames()))
+
+	for scope, collections := range auth.Collections {
+		for collection, _ := range collections {
+			channels := princ.CollectionChannels(scope, collection).Copy()
+			for name := range user.RoleNames() {
+				role, err := auth.GetRole(name)
+				if err != nil {
+					return err
+				} else if role != nil {
+					roles = append(roles, role)
+				}
+			}
```

Comment:
Isn't this only required once? Roles are associated with the database, not a specific collection